objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/data_detection.py

```diff
@@ -0,0 +1,345 @@
+        ----------
+        metadata_csv_file_path
+            Path of the CSV file containing the metadata of the pictures,
+            it's used to keep track of which files had to be detected.
+        """
+        csv_path = pathlib.Path(metadata_csv_file_path)
+        relative_path = csv_path.relative_to(self.images_folder)
+        images_path = self.images_folder / relative_path.parent
+        with open(metadata_csv_file_path) as frame_metadata_file:
+            images_moved_count = 0
+            reader = csv.reader(frame_metadata_file)
+            _ = next(reader)
+            for idx, row in enumerate(reader):
+                image_file_name = get_img_name_from_csv_row(csv_path, row)
+                image_full_path = images_path / image_file_name
+                if os.path.isfile(image_full_path):
+                    image_subfolder_and_name = os.path.relpath(
+                        image_full_path, self.images_folder
+                    )
+                    image_destination_full_path = os.path.join(
+                        self.training_mode_destination_path,
+                        image_subfolder_and_name,
+                    )
+                    move_file(image_full_path, image_destination_full_path)
+                    images_moved_count += 1
+        metadata_csv_subfolder_and_name = os.path.relpath(csv_path, self.images_folder)
+        metadata_csv_destination_file_path = os.path.join(
+            self.training_mode_destination_path, metadata_csv_subfolder_and_name
+        )
+        move_file(csv_path, metadata_csv_destination_file_path)
+        logger.info(
+            f"Training mode on: Moved {images_moved_count} images from {metadata_csv_file_path}"
+        )
+
+    @log_execution_time
+    def _detect_and_blur_image(
+        self,
+        image_file_name: pathlib.Path,
+        image_full_path: pathlib.Path,
+        csv_path: pathlib.Path,
+        detections_path: pathlib.Path,
+    ):
+        """Loads the image, resizes it, detects containers and blur sensitive data.
+
+        Parameters
+        ----------
+        image_file_name : pathlib.Path
+            File name of the image
+        image_full_path : pathlib.Path
+            Path of the image
+        csv_path : pathlib.Path
+            Path where the csv metadata file is stored. For example:
+            /detections/folder1/file1.csv
+        detections_path : pathlib.Path
+            Path of detections excluding the file. For example:
+            /detections/folder1
+
+        Returns
+        -------
+        int
+            Count of detected target objects
+        """
+        logger.debug(f"Detecting and blurring: {image_file_name}")
+
+        image = InputImage(image_full_path=str(image_full_path))
+        image.resize(output_image_size=self.output_image_size)
+
+        if self.defisheye_flag:
+            image.defisheye(defisheye_params=self.defisheye_params)
+
+        self.inference_params["source"] = image
```

Comment:
I don't think this works, should probably be `self.inference_params["source"] = image.image` (or create a getter method, or some other python funk that pretends the InputImage is in fact an np.ndarray).